### PR TITLE
fix search bar bug

### LINKED ISSUE #50064: Rust documentation search bar bug
The search bar in the documentation (on both https://doc.rust-lang.org/std/ and https://docs.rs) does not provide the results of a research if the searched word has been researched just before.

Steps to reproduce:
- In the search bar of https://doc.rust-lang.org/std, press a.
- Wait for the result page to appear.
- Press backspace.
- Wait for the original page to come back.
- Press a.
- The result page does not come back.

This has been observed on stable, beta and nightly, and works with any word, as long as it is written fast enough, and no intermediate search result are shown.

@GuillaumeGomez

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #50064.

r? @QuietMisdreavus

### CODE REVIEW

File: src/librustdoc/html/static/main.js [hunk 1]
```diff
@@ -1369,13 +1369,17 @@
...
-            var query = getQuery(document.getElementsByClassName('search-input')[0].value);
+            var search_input = document.getElementsByClassName('search-input')[0];
+            var query = getQuery(search_input.value);
 
             if (e) {
                 e.preventDefault();
             }
 
             if (!query.query || query.id === currentResults) {
+                if (query.query.length > 0) {
```
Comment (QuietMisdreavus) [inline src/librustdoc/html/static/main.js]: Will this bite us if this block is entered from `!query.query`? Can `query.query` be undefined here?

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/main.js]: Based on [this code](https://github.com/rust-lang/rust/blob/master/src/librustdoc/html/static/main.js#L1056), I'd say no. This is why I'm complaining about such conditions: it's difficult to understand what we're checking. I think the "real" condition here is `if (query.query.length === 0 || query.id === currentResults) {`.

**Comment by QuietMisdreavus** [comment]
Makes sense. Thanks for the clarification!

@bors r+ rollup

**Comment by GuillaumeGomez** [comment]
Fixed merge conflict.

@bors r=QuietMisdreavus

**Pull request merged**